pkg/agent/types.go

```diff
@@ -12,6 +14,19 @@ type KeyVaultRef struct {
 	SecretVersion string     `json:"secretVersion,omitempty"`
 }
 
+// NodeBootstrappingConfiguration represents configurations used to generate CSE for node
```

Comment:
not just cse, but also cloud init. myabe just be general and don't mention the detail